torchtune/_cli/hf_upload/upload.py

```diff
@@ -57,24 +57,23 @@ def upload_to_hf_hub(args: Any):
         )
 
     # Initialize Hugging Face API
-    api = HfApi()
+    api = HfApi(token=args.hf_token)
 
-    # Create a new private repository on the Hub
-    repo_url = api.create_repo(
-        args.repo_name, private=args.private, token=args.hf_token, exist_ok=True
-    )
-
-    # Clone the repository
-    repo = Repository(args.repo_name, clone_from=repo_url, use_auth_token=args.hf_token)
-    repo.git_pull()
-
-    # Copy files to the repository
-    file_paths = [args.model_path, args.readme_path, args.license_path]
-    add_files_to_repo(api, args.hf_token, args.hf_username, args.repo_name, file_paths)
+    # Create a new repository on the Hub
+    repo_url = api.create_repo(args.repo_id, private=args.private, exist_ok=True)
 
-    # Commit and push the files to the repository
-    repo.commit("Initial commit with model, README, and License")
-    repo.push_to_hub()
+    # Upload readme, license and weights
+    api.create_commit(
+        repo_id=repo_url.repo_id,
+        operations=[
+            CommitOperationAdd(
+                path_in_repo=os.path.basename(file_path), path_or_fileobj=file_path
+            )
+            for file_path in [args.model_path, args.readme_path, args.license_path]
+            if file_path and os.path.exists(file_path)
```

Comment:
We should ensure we have the README and licenses for sure to comply with legal. So if something doesn't pass this `if file path and os.exists` check, let's bail out with an error just to be super careful?